app/src/main/java/com/example/hris/ui/adapters/LeavesListAdapter.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.hris.ui.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.hris.R
+import com.example.hris.convertToNoDecimalString
+import com.example.hris.databinding.LeavesListBinding
+import com.example.hris.getNumberOfDaysInBetween
+import com.example.hris.model.Leaves
+import com.example.hris.ui.viewmodels.leaves.LeavesViewModel
+
+class LeavesListAdapter(
+    private val list: List<Leaves>,
+    private val viewModel: LeavesViewModel
```

Comment:
You're still passing the `ViewModel` to the `LeavesListAdapter`. The adapter should just have a callback to the `LeavesFragment`, then the `LeavesFragment` should be the one communicating with `LeavesViewModel`.